--- a/test/automated/Makefile.in
+++ b/test/automated/Makefile.in
@@ -116,7 +116,8 @@ endif
 	$(emacs) -l ert -l $$loadfile \
 	  --eval "(ert-run-tests-batch-and-exit ${SELECTOR_ACTUAL})" ${WRITE_LOG}

-ELFILES = $(sort $(wildcard ${srcdir}/*.el))
+#ELFILES = $(sort $(wildcard ${srcdir}/*.el))
+ELFILES = $(patsubst %tramp-tests.el,,$(sort $(wildcard ${srcdir}/*.el)))
 LOGFILES = $(patsubst %.el,%.log,$(notdir ${ELFILES}))
 TESTS = ${LOGFILES:.log=}

