diff --git a/src/Test/IOSpec/STM.hs b/src/Test/IOSpec/STM.hs
index 357c646..458656b 100644
--- a/src/Test/IOSpec/STM.hs
+++ b/src/Test/IOSpec/STM.hs
@@ -23,6 +23,7 @@ import Test.IOSpec.Types
 import Data.Dynamic
 import Data.Maybe (fromJust)
 import Control.Monad.State
+import Control.Applicative
 import Control.Monad (ap)
 
 -- The 'STMS' data type and its instances.
diff --git a/src/Test/IOSpec/Types.hs b/src/Test/IOSpec/Types.hs
index 8b85fe8..381a092 100644
--- a/src/Test/IOSpec/Types.hs
+++ b/src/Test/IOSpec/Types.hs
@@ -15,6 +15,7 @@ module Test.IOSpec.Types
   , inject
   ) where
 
+import Control.Applicative
 import Control.Monad (ap)
 
 -- | A value of type 'IOSpec' @f@ @a@ is either a pure value of type @a@
diff --git a/src/Test/IOSpec/VirtualMachine.hs b/src/Test/IOSpec/VirtualMachine.hs
index 093db46..542bee9 100644
--- a/src/Test/IOSpec/VirtualMachine.hs
+++ b/src/Test/IOSpec/VirtualMachine.hs
@@ -43,6 +43,7 @@ import Data.List
 import qualified Data.Stream as Stream
 import Test.IOSpec.Types
 import Test.QuickCheck
+import Control.Applicative
 import Control.Monad (ap)
 
 type Data         = Dynamic
