diff --git c/pkgs/applications/editors/emacs/default.nix w/pkgs/applications/editors/emacs/default.nix
index 1ac714c15575..a564fc836516 100644
--- c/pkgs/applications/editors/emacs/default.nix
+++ w/pkgs/applications/editors/emacs/default.nix
@@ -13,6 +13,7 @@ lib.makeScope pkgs.newScope (
       inherit lib;
       inherit (pkgs)
         fetchFromBitbucket
+        fetchFromGitHub
         fetchzip
         ;
     };
@@ -31,6 +32,6 @@ lib.makeScope pkgs.newScope (
       withPgtk = true;
     };
 
-    emacs29-macport = callPackage (self.sources.emacs29-macport) inheritedArgs;
+    emacs30-macport = callPackage (self.sources.emacs30-macport) inheritedArgs;
   }
 )
diff --git c/pkgs/applications/editors/emacs/macport-stdbool.patch w/pkgs/applications/editors/emacs/macport-stdbool.patch
new file mode 100644
index 000000000000..5d3e71ed3942
--- /dev/null
+++ w/pkgs/applications/editors/emacs/macport-stdbool.patch
@@ -0,0 +1,12 @@
+diff --git a/src/conf_post.h b/src/conf_post.h
+index e3272278832..01d3bcf9a81 100644
+--- a/src/conf_post.h
++++ b/src/conf_post.h
+@@ -58,6 +58,7 @@ along with GNU Emacs.  If not, see <https://www.gnu.org/licenses/>.  */
+ #if NS_IMPL_GNUSTEP || defined __MINGW32__
+ typedef unsigned int bool_bf;
+ #else
++#include <stdbool.h> /* Clang in C mode does not automatically define bool unless you explicitly include that header */
+ typedef bool bool_bf;
+ #endif
+ 
diff --git c/pkgs/applications/editors/emacs/macport_noescape_noop.h w/pkgs/applications/editors/emacs/macport_noescape_noop.h
index 64c65bd90f88..66accdd04faf 100644
--- c/pkgs/applications/editors/emacs/macport_noescape_noop.h
+++ w/pkgs/applications/editors/emacs/macport_noescape_noop.h
@@ -1,3 +1,7 @@
+#if __has_include("config.h")
+#include "config.h"
+#endif
+
 #ifndef NOESCAPE_NOOP_H_
 #define NOESCAPE_NOOP_H_
 
diff --git c/pkgs/applications/editors/emacs/make-emacs.nix w/pkgs/applications/editors/emacs/make-emacs.nix
index b9f75464c793..2106053bbf18 100644
--- c/pkgs/applications/editors/emacs/make-emacs.nix
+++ w/pkgs/applications/editors/emacs/make-emacs.nix
@@ -13,7 +13,7 @@
   Xaw3d,
   acl,
   alsa-lib,
-  apple-sdk,
+  apple-sdk_14,
   autoreconfHook,
   cairo,
   dbus,
@@ -190,7 +190,7 @@ mkDerivation (finalAttrs: {
                 ]
                 ++ lib.optionals stdenv.hostPlatform.isDarwin [
                   # The linker needs to know where to find libSystem on Darwin.
-                  "${apple-sdk.sdkroot}/usr/lib"
+                  "${apple-sdk_14.sdkroot}/usr/lib"
                 ]
               )
             )
diff --git c/pkgs/applications/editors/emacs/sources.nix w/pkgs/applications/editors/emacs/sources.nix
index d3fb3da3b599..629420d960ad 100644
--- c/pkgs/applications/editors/emacs/sources.nix
+++ w/pkgs/applications/editors/emacs/sources.nix
@@ -1,6 +1,7 @@
 {
   lib,
   fetchFromBitbucket,
+  fetchFromGitHub,
   fetchzip,
 }:
 
@@ -32,8 +33,8 @@ let
             }
           );
           "macport" = (
-            fetchFromBitbucket {
-              owner = "mituharu";
+            fetchFromGitHub {
+              owner = "jdtsmith";
               repo = "emacs-mac";
               inherit rev hash;
             }
@@ -88,7 +89,9 @@ let
               matthewbauer
               panchoh
             ];
-            "macport" = with lib.maintainers; [ ];
+            "macport" = with lib.maintainers; [
+              lactose
+            ];
           }
           .${variant};
         platforms =
@@ -117,25 +120,13 @@ in
     ];
   });
 
-  emacs29-macport = import ./make-emacs.nix (mkArgs {
+  emacs30-macport = import ./make-emacs.nix (mkArgs {
     pname = "emacs-mac";
-    version = "29.4";
+    version = "30.2.50";
     variant = "macport";
-    rev = "emacs-29.4-mac-10.1";
-    hash = "sha256-8OQ+fon9tclbh/eUJ09uqKfMaz9M77QnLIp2R8QB6Ic=";
-    patches = fetchpatch: [
-      # CVE-2024-53920
-      (fetchpatch {
-        url = "https://gitweb.gentoo.org/proj/emacs-patches.git/plain/emacs/29.4/07_all_trusted-content.patch?id=f24370de4de0a37304958ec1569d5c50c1745b7f";
-        hash = "sha256-zUWM2HDO5MHEB5fC5TCUxzmSafMvXO5usRzCyp9Q7P4=";
-      })
-
-      # CVE-2025-1244
-      (fetchpatch {
-        url = "https://gitweb.gentoo.org/proj/emacs-patches.git/plain/emacs/29.4/06_all_man.patch?id=f24370de4de0a37304958ec1569d5c50c1745b7f";
-        hash = "sha256-Vdf6GF5YmGoHTkxiD9mdYH0hgvfovZwrqYN1NQ++U1w=";
-      })
-    ];
+    rev = "emacs-macs-30_1_exp";
+    hash = "sha256-dDkKSUfxPCWiF+MXQBY21npjZ6LZL6IG6GPx5AptegI=";
+    patches = _: [ ./macport-stdbool.patch ];
 
     meta.knownVulnerabilities = [ ];
   });
diff --git c/pkgs/top-level/aliases.nix w/pkgs/top-level/aliases.nix
index c47470549d1e..4d9ddd5ec72c 100644
--- c/pkgs/top-level/aliases.nix
+++ w/pkgs/top-level/aliases.nix
@@ -829,6 +829,7 @@ mapAliases {
   emacs29-gtk3 = throw "Emacs 29 is removed due to CVEs which are fixed in Emacs 30"; # Added 2025-03-03
   emacs29-nox = throw "Emacs 29 is removed due to CVEs which are fixed in Emacs 30"; # Added 2025-03-03
   emacs29-pgtk = throw "Emacs 29 is removed due to CVEs which are fixed in Emacs 30"; # Added 2025-03-03
+  emacs29-macport = throw "Emacs 29 is removed due to CVEs which are fixed in Emacs 30"; # Added 2025-04-02
   emacsMacport = emacs-macport; # Added 2023-08-10
   emacsNativeComp = emacs; # Added 2022-06-08
   emacsWithPackages = throw "'emacsWithPackages' has been renamed to/replaced by 'emacs.pkgs.withPackages'"; # Converted to throw 2024-10-17
diff --git c/pkgs/top-level/all-packages.nix w/pkgs/top-level/all-packages.nix
index 0f58b6d76629..ddd639580f3c 100644
--- c/pkgs/top-level/all-packages.nix
+++ w/pkgs/top-level/all-packages.nix
@@ -11492,10 +11492,10 @@ with pkgs;
     emacs30-nox
     emacs30-pgtk
 
-    emacs29-macport
+    emacs30-macport
     ;
 
-  emacs-macport = emacs29-macport;
+  emacs-macport = emacs30-macport;
   emacs = emacs30;
   emacs-gtk = emacs30-gtk3;
   emacs-nox = emacs30-nox;
