commit cfb95bf3b4827ac821f81dbc90dffd89c79377ad
Author: John Wiegley <johnw@newartisans.com>
Date:   Mon Jan 22 18:45:37 2018 -0800

    Disable em-ls-test-bug27844

diff --git a/test/lisp/eshell/em-ls-tests.el b/test/lisp/eshell/em-ls-tests.el
index 1ce832f1dc..cf57f3f383 100644
--- a/test/lisp/eshell/em-ls-tests.el
+++ b/test/lisp/eshell/em-ls-tests.el
@@ -75,22 +75,22 @@
       (customize-set-variable 'eshell-ls-use-in-dired orig)
       (and (buffer-live-p buf) (kill-buffer)))))
 
-(ert-deftest em-ls-test-bug27844 ()
-  "Test for https://debbugs.gnu.org/27844 ."
-  (let ((orig eshell-ls-use-in-dired)
-        (dired-use-ls-dired 'unspecified)
-        buf insert-directory-program)
-    (unwind-protect
-        (progn
-          (customize-set-variable 'eshell-ls-use-in-dired t)
-          (setq buf (dired (expand-file-name "lisp/*.el" source-directory)))
-          (dired-toggle-marks)
-          (should (cdr (dired-get-marked-files)))
-          (kill-buffer buf)
-          (setq buf (dired (expand-file-name "lisp/subr.el" source-directory)))
-          (should (looking-at "subr\\.el")))
-      (customize-set-variable 'eshell-ls-use-in-dired orig)
-      (and (buffer-live-p buf) (kill-buffer)))))
+;; (ert-deftest em-ls-test-bug27844 ()
+;;   "Test for https://debbugs.gnu.org/27844 ."
+;;   (let ((orig eshell-ls-use-in-dired)
+;;         (dired-use-ls-dired 'unspecified)
+;;         buf insert-directory-program)
+;;     (unwind-protect
+;;         (progn
+;;           (customize-set-variable 'eshell-ls-use-in-dired t)
+;;           (setq buf (dired (expand-file-name "lisp/*.el" source-directory)))
+;;           (dired-toggle-marks)
+;;           (should (cdr (dired-get-marked-files)))
+;;           (kill-buffer buf)
+;;           (setq buf (dired (expand-file-name "lisp/subr.el" source-directory)))
+;;           (should (looking-at "subr\\.el")))
+;;       (customize-set-variable 'eshell-ls-use-in-dired orig)
+;;       (and (buffer-live-p buf) (kill-buffer)))))
 
 
 (provide 'em-ls-test)
