--- a/haskell-mode.el
+++ b/haskell-mode.el
@@ -1043,7 +1043,7 @@ To be added to `flymake-init-create-temp-buffer-copy'."
                            (split-string haskell-saved-check-command))))
     (list (car checker-elts)
           (append (cdr checker-elts)
-                  (list (flymake-init-create-temp-buffer-copy
+                  (list (flymake-proc-init-create-temp-buffer-copy
                          'flymake-create-temp-inplace))))))

 (add-to-list 'flymake-allowed-file-name-masks '("\\.l?hs\\'" haskell-flymake-init))
