--- a/hyperbole.el
+++ b/hyperbole.el
@@ -280,12 +280,12 @@ Entry format is: (key-description key-sequence key-binding)."
     ;; button renames without invoking the Hyperbole menu.
     ;;
     ;; Don't override local bindings of this key.
-    (hkey-maybe-global-set-key "\C-c\C-r" 'hui:ebut-rename t)
+    ;; (hkey-maybe-global-set-key "\C-c\C-r" 'hui:ebut-rename t)
     ;;
     ;; Binds {C-c RET} to select larger and larger synctactical units in a
     ;; buffer when invoked repeatedly, showing in the minibuffer the type
     ;; of unit selected each time.
-    (hkey-maybe-global-set-key "\C-c\C-m" 'hui-select-thing)
+    ;; (hkey-maybe-global-set-key "\C-c\C-m" 'hui-select-thing)
     ;;
     ;; Binds {C-c \} to interactively manage windows and frames.
     (hkey-maybe-global-set-key "\C-c\\" 'hycontrol-enable-windows-mode)
    Re-enable Hyperbole after a workaround
@@ -262,13 +262,13 @@ Entry format is: (key-description key-sequence key-binding)."
     ;; commands from the keyboard.  This is most useful for rapidly creating
     ;; Hyperbole link buttons from the keyboard without invoking the Hyperbole
     ;; menu.  Only works if Hyperbole is run under a window system.
-    (when (hyperb:window-system)
-      (if (eq (global-key-binding "\M-o") 'facemenu-keymap)
-	  ;; Override facemenu package that adds a keymap on M-o,
-	  ;; since this binding is more important to Hyperbole
-	  ;; users.
-	  (hkey-global-set-key "\M-o" 'hkey-operate)
-	(hkey-maybe-global-set-key "\M-o" 'hkey-operate)))
+    ;; (when (hyperb:window-system)
+    ;;   (if (eq (global-key-binding "\M-o") 'facemenu-keymap)
+    ;;       ;; Override facemenu package that adds a keymap on M-o,
+    ;;       ;; since this binding is more important to Hyperbole
+    ;;       ;; users.
+    ;;       (hkey-global-set-key "\M-o" 'hkey-operate)
+    ;;     (hkey-maybe-global-set-key "\M-o" 'hkey-operate)))
     ;;
     ;; Binds {C-c @} to created a user-specified sized grid of windows
     ;; displaying different buffers.
--- a/hmouse-drv.el
+++ b/hmouse-drv.el
@@ -431,10 +431,10 @@ Return non-nil iff associated documentation is found."
   (interactive)
   (hkey-help 'assist))

-;; Overload help-mode quit-window function to support Hyperbole
-;; hkey--wconfig window configurations.
-(unless (eq (symbol-function #'quit-window) #'hkey-help-hide)
-  (defalias 'hkey-quit-window (hypb:function-copy #'quit-window)))
+;; ;; Overload help-mode quit-window function to support Hyperbole
+;; ;; hkey--wconfig window configurations.
+;; (unless (eq (symbol-function #'quit-window) #'hkey-help-hide)
+;;   (defalias 'hkey-quit-window (hypb:function-copy #'quit-window)))

 ;;;###autoload
 (defun hkey-help-hide (&optional kill window)
@@ -451,7 +451,7 @@ details."
       (hkey-quit-window kill window)))
   (setq hkey--wconfig nil))

-(defalias 'quit-window 'hkey-help-hide)
+;; (defalias 'quit-window 'hkey-help-hide)

 ;; Newer versions of Emacs define this variable but older versions,
 ;; e.g. Emacs 22, do not.  Calls to the `with-help-buffer' macro
