--- a/multi-term.el
+++ b/multi-term.el
@@ -345,13 +345,13 @@ So please turn on this option if you want to skip `multi-term' dedicated window

 Default is nil."
   :type 'boolean
-  :set (lambda (symbol value)
-         (set symbol value)
-         ;; ad-advised-definition-p no longer exists on Emacs 24.4 as of 2014-01-03.
-         (when (fboundp 'multi-term-dedicated-handle-other-window-advice)
-           (if (fboundp 'ad-advised-definition-p)
-               (when (ad-advised-definition-p 'other-window)
-                 (multi-term-dedicated-handle-other-window-advice value))
-             (when (ad-is-advised 'other-window)
-               (multi-term-dedicated-handle-other-window-advice value)))))
+  ;; :set (lambda (symbol value)
+  ;;        (set symbol value)
+  ;;        ;; ad-advised-definition-p no longer exists on Emacs 24.4 as of 2014-01-03.
+  ;;        (when (fboundp 'multi-term-dedicated-handle-other-window-advice)
+  ;;          (if (fboundp 'ad-advised-definition-p)
+  ;;              (when (ad-advised-definition-p 'other-window)
+  ;;                (multi-term-dedicated-handle-other-window-advice value))
+  ;;            (when (ad-is-advised 'other-window)
+  ;;              (multi-term-dedicated-handle-other-window-advice value)))))
   :group 'multi-term)

 (defcustom multi-term-dedicated-select-after-open-p nil
