--- a/password-store.el
+++ b/password-store.el
@@ -217,11 +217,17 @@
 (defun password-store-insert (entry password)
   "Insert a new ENTRY containing PASSWORD."
   (interactive (list (read-string "Password entry: ")
+                     (read-string "Login: ")
                      (read-passwd "Password: " t)))
-  (message "%s" (shell-command-to-string (format "echo %s | %s insert -m -f %s"
-                                            (shell-quote-argument password)
-                                            password-store-executable
-                                            (shell-quote-argument entry)))))
+  (message "%s" (shell-command-to-string
+                 (if (string= "" login)
+                     (format "echo %s | %s insert -m -f %s"
+                             (shell-quote-argument password)
+                             password-store-executable
+                             (shell-quote-argument entry))
+                   (format "echo -e '%s\nlogin: %s' | %s insert -m -f %s"
+                           password login password-store-executable
+                           (shell-quote-argument entry))))))
 
 ;;;###autoload
 (defun password-store-generate (entry &optional password-length)
