diff -U3 -w -r /nix/store/d6j6b3q4611ihv2vid3l5sm432rzybph-source/contrib/emacs/password-store.el ./contrib/emacs/password-store.el
--- /nix/store/d6j6b3q4611ihv2vid3l5sm432rzybph-source/contrib/emacs/password-store.el	1969-12-31 16:00:01.000000000 -0800
+++ ./contrib/emacs/password-store.el	2018-01-29 15:03:11.000000000 -0800
@@ -224,15 +224,20 @@
   (message "%s" (password-store--run-init (split-string gpg-id))))

 ;;;###autoload
-(defun password-store-insert (entry password)
+(defun password-store-insert (entry login password)
   "Insert a new ENTRY containing PASSWORD."
   (interactive (list (read-string "Password entry: ")
+                     (read-string "Login: ")
                      (read-passwd "Password: " t)))
   (message "%s" (shell-command-to-string
+                 (if (string= "" login)
                  (format "echo %s | %s insert -m -f %s"
                          (shell-quote-argument password)
                          password-store-executable
-                         (shell-quote-argument entry)))))
+                             (shell-quote-argument entry))
+                   (format "echo -e '%s\nlogin: %s' | %s insert -m -f %s"
+                           password login password-store-executable
+                           (shell-quote-argument entry))))))
 
 ;;;###autoload
 (defun password-store-generate (entry &optional password-length)
