diff --git a/server/poppler-hack.cc b/server/poppler-hack.cc
index 427f9df..ca92716 100644
--- a/server/poppler-hack.cc
+++ b/server/poppler-hack.cc
@@ -54,7 +54,7 @@ GType poppler_annot_markup_get_type (void) G_GNUC_CONST;
   // This function does not modify its argument s, but for
   // compatibility reasons (e.g. getLength in GooString.h before 2015)
   // with older poppler code, it can't be declared as such.
-  char *_xpoppler_goo_string_to_utf8(/* const */ GooString *s)
+  char *_xpoppler_goo_string_to_utf8(const GooString *s)
   {
     char *result;
 
@@ -88,7 +88,7 @@ GType poppler_annot_markup_get_type (void) G_GNUC_CONST;
   // Set the rectangle of an annotation.  It was first added in v0.26.
   void xpoppler_annot_set_rectangle (PopplerAnnot *a, PopplerRectangle *rectangle)
   {
-    GooString *state = (GooString*) a->annot->getAppearState ();
+    const GooString *state = (GooString*) a->annot->getAppearState ();
     char *ustate = _xpoppler_goo_string_to_utf8 (state);
 
     a->annot->setRect (rectangle->x1, rectangle->y1,
